src/pages/analytics/analytics.ts

```diff
@@ -0,0 +1,145 @@
+import { Component, ViewChild, ElementRef } from '@angular/core';
+import { Expense, Category } from '../../models/expense-model';
+import { BudgetService } from '../../services/budget-service';
+import { compareDesc } from 'date-fns';
+import { Chart } from 'chart.js';
+import 'chartjs-plugin-labels';
+
+@Component({
+  selector: 'page-analytics',
+  templateUrl: 'analytics.html'
+})
+export class AnalyticsPage {
+    public expenses: Expense[] = [];
+    public categories: Category[] = this._budgetService.getCategories();
+    public myChart = [];
+    public chartData = new Map<string,number>();
+
+  constructor(private _budgetService: BudgetService) {
+  }
+
+  public getSortedExpenses() {
+    let sortedExpenses = this._budgetService.getBudget().expenses.sort((x, y) => {
+      return compareDesc(x.date, y.date);
+    });
+    sortedExpenses.forEach(e => e.date = new Date(e.date));
+    this.expenses = sortedExpenses;
```

Comment:
given that the name is `getSortedExpenses` should we return the expenses instead of it modifying the global expenses prop? Makes it more explicit on what to return.